src/iterateJsdoc.js

```diff
@@ -33,41 +56,52 @@ const curryUtils = (functionNode, jsdoc, tagNamePreference, additionalTagNames,
     return allowOverrideWithoutParam;
   };
 
-  return utils;
-};
+  utils.isImplementsAllowedWithoutParam = () => {
+    return allowImplementsWithoutParam;
+  };
 
-export const parseComment = (commentNode, indent) => {
-  // Preserve JSDoc block start/end indentation.
-  return commentParser(indent + '/*' + commentNode.value + indent + '*/', {
-    // @see https://github.com/yavorskiy/comment-parser/issues/21
-    parsers: [
-      commentParser.PARSERS.parse_tag,
-      commentParser.PARSERS.parse_type,
-      (str, data) => {
-        if (_.includes(['return', 'returns'], data.tag)) {
-          return null;
-        }
+  utils.isAugmentsExtendsAllowedWithoutParam = () => {
+    return allowAugmentsExtendsWithoutParam;
+  };
 
-        return commentParser.PARSERS.parse_name(str, data);
-      },
-      commentParser.PARSERS.parse_description
-    ]
-  })[0] || {};
+  utils.classHasTag = function (tagName) {
+    const greatGrandParent = ancestors.slice(-3)[0];
+    const greatGrandParentValue = greatGrandParent && sourceCode.getFirstToken(greatGrandParent).value;
+
+    if (greatGrandParentValue === 'class') {
+      const classJsdocNode = sourceCode.getJSDocComment(greatGrandParent);
+      const indent = _.repeat(' ', classJsdocNode.loc.start.column);
+      const classJsdoc = parseComment(classJsdocNode, indent);
+
+      if (jsdocUtils.hasTag(classJsdoc, tagName)) {
+        return true;
+      }
+    }
+
+    return false;
+  };
+
+  return utils;
 };
 
+export {parseComment};
+
 export default (iterator) => {
   return (context) => {
     const sourceCode = context.getSourceCode();
     const tagNamePreference = _.get(context, 'settings.jsdoc.tagNamePreference') || {};
     const additionalTagNames = _.get(context, 'settings.jsdoc.additionalTagNames') || {};
     const allowOverrideWithoutParam = Boolean(_.get(context, 'settings.jsdoc.allowOverrideWithoutParam'));
+    const allowImplementsWithoutParam = Boolean(_.get(context, 'settings.jsdoc.allowImplementsWithoutParam'));
+    const allowAugmentsExtendsWithoutParam = Boolean(_.get(context, 'settings.jsdoc.allowAugmentsExtendsWithoutParam'));
 
     const checkJsdoc = (functionNode) => {
       const jsdocNode = sourceCode.getJSDocComment(functionNode);
 
       if (!jsdocNode) {
         return;
       }
+      const ancestors = context.getAncestors();
```

Comment:
style: add a space before ancestors.

Reply:
I guess you mean a line break, right? It's at the right indent...